src/QueryPipeline/QueryPlanResourceHolder.cpp

```diff
@@ -4,7 +4,7 @@
 namespace DB
 {
 
-QueryPlanResourceHolder & QueryPlanResourceHolder::append(QueryPlanResourceHolder && rhs) noexcept
+QueryPlanResourceHolder & QueryPlanResourceHolder::append(const QueryPlanResourceHolder & rhs) noexcept
```

Comment:
Seems odd to move from a move reference to a const reference (also below). Although we should use the fact that's movable when copying /  extracting the resources.